msibi/utils/conv.py

```diff
@@ -0,0 +1,207 @@
+import MDAnalysis as mda
+import gsd.hoomd
+import numpy as np
+
+def _mda_check(topology, trajectory):
+    """
+    MDAnalysis input check:
+      - Topology must load.
+      - Bonds required if any residue has >=2 atoms.
+      - Angles required if any residue has >=3 atoms.
+      - If trajectory given, it must load with the topology and atom counts must match.
+    """
+    out = {
+        "valid_topology": False,
+        "need_bonds": False,
+        "need_angles": False,
+        "has_bonds": False,
+        "has_angles": False,
+        "valid_trajectory": False,
+        "match": False,
+    }
+
+    if not topology:
+        return out
+
+    # Load topology only
+    try:
+        u_top = mda.Universe(topology)
+        out["valid_topology"] = True
+    except:
+        return out
+
+    # --- Decide requirements based on residue sizes
+    max_atoms_in_residue = max((len(res.atoms) for res in u_top.residues), default=0)
+    out["need_bonds"] = max_atoms_in_residue >= 2
+    out["need_angles"] = max_atoms_in_residue >= 3
+
+    # --- Check what's present
+    out["has_bonds"] = len(u_top.bonds) > 0
+    out["has_angles"] = len(u_top.angles) > 0
+
+    # --- Trajectory check
+    if trajectory:
+        try:
+            u = mda.Universe(topology, trajectory)
+            out["valid_trajectory"] = True
+            out["match"] = (u.atoms.n_atoms == u_top.atoms.n_atoms)
+        except:
```

Comment:
Similar idea here as what I said above. What error do we expect to be given in this case where we want to pass?